ui-spacetimechart/src/stories/paths-interactions.stories.tsx

```diff
@@ -0,0 +1,244 @@
+import React, { FC, useState } from 'react';
+import type { Meta } from '@storybook/react';
+
+import { OPERATIONAL_POINTS, PATHS } from './assets/paths';
+import { SpaceTimeChart, PathLayer } from '../';
+import { HoveredItem, PathData, Point } from '../core/types';
+import { getDiff } from '../utils/vectors';
+import { X_ZOOM_LEVEL, Y_ZOOM_LEVEL, zoom } from './utils';
+
+import './tailwind-mockup.css';
+
+function delayPath<T extends PathData>(path: T, newTimeOrigin: number): T {
+  const delay = newTimeOrigin - path.points[0].time;
+  return {
+    ...path,
+    points: path.points.map((point) => ({ ...point, time: point.time + delay })),
+  };
+}
+
+/**
+ * This story aims at showcasing how to manipulate paths in a SpaceTimeChart.
+ */
+const Wrapper: FC<{
+  dragPaths: boolean;
+  multiSelection: boolean;
+  spaceScaleType: 'linear' | 'proportional';
+  pickingTolerance: number;
+}> = ({ dragPaths, pickingTolerance, multiSelection, spaceScaleType }) => {
+  const [paths, setPaths] = useState(PATHS);
+  const [state, setState] = useState<{
+    xOffset: number;
+    yOffset: number;
+    xZoomLevel: number;
+    yZoomLevel: number;
+    selection: Set<number> | null;
+    panTarget:
+      | null
+      | { type: 'stage'; initialOffset: Point }
+      | { type: 'items'; initialTimeOrigins: Record<number, number> };
+    hoveredPath: HoveredItem | null;
+  }>({
+    xOffset: 0,
+    yOffset: 0,
+    xZoomLevel: X_ZOOM_LEVEL,
+    yZoomLevel: Y_ZOOM_LEVEL,
+    selection: null,
+    panTarget: null,
+    hoveredPath: null,
+  });
+
+  return (
+    <div className="inset-0">
+      <SpaceTimeChart
+        className={cx(
+          'inset-0 absolute p-0 m-0',
+          state.panTarget && 'cursor-grabbing',
+          state.hoveredPath && 'cursor-pointer'
+        )}
+        operationalPoints={OPERATIONAL_POINTS}
+        spaceOrigin={0}
+        spaceScales={OPERATIONAL_POINTS.slice(0, -1).map((point, i) => ({
+          from: point.position,
+          to: OPERATIONAL_POINTS[i + 1].position,
+          ...(spaceScaleType === 'linear'
+            ? { size: 50 * state.yZoomLevel }
+            : { coefficient: 150 / state.yZoomLevel }),
+        }))}
+        timeOrigin={+new Date('2024/04/02')}
+        timeScale={60000 / state.xZoomLevel}
+        xOffset={state.xOffset}
+        yOffset={state.yOffset}
+        onClick={({ event }) => {
+          const { hoveredPath, selection, panTarget } = state;
+
+          // Skip events when something is being dragged or panned:
+          if (panTarget) return;
+
+          // Unselect everything when clicking stage:
+          if (!hoveredPath) setState((state) => ({ ...state, selection: null }));
+          // Select item when nothing is selected:
+          else if (!selection)
+            setState({
+              ...state,
+              selection: new Set([hoveredPath.index]),
+            });
+          // Handle single selection:
+          else if (!multiSelection || !event.ctrlKey) {
```

Comment:
So we handle multiselection only if the multiselection flag is true and the ctrlKey is down ?

Reply:
In that story, yes. But if we decide to handle multi-selection differently in OSRD, it remains possible. Here, it's just an implementation example.